tests/seechange_config_test.yaml

```diff
@@ -5,15 +5,19 @@ overrides:
 augments:
   - local_augments.yaml
 
+path:
+  data_root: 'tests/temp_data'
+  data_temp: 'tests/temp_data'
+
 db:
   host: seechange_postgres
 
 archive:
   archive_url: http://archive:8080/
   verify_cert: false
   path_base: test/
-  local_read_dir: null
-  local_write_dir: null
+  local_read_dir: /storage/base/
```

Comment:
...hmm, this one alarms me a little bit.  I will have to look at how archive is implemented, but the idea the archive is contacted via url XOR the archive is read/written from local files.  Having a local read/write dir *and* a archive_url may lead the logic to not doing what we think it's doing.

I would prefer to document that local_read_dir and local_write_dir can be non-null if and only if archive_url is null. (And, then, perhaps, put a check and exception if it finds it otherwise.)  Right now, it's not obvious how the archive would be accessed.

I'd rather have something ad-hoc that's explicitly for the test environment than reusing the archive configuration itself, so we can be sure that the archive code isn't misusing the configuration here when we test it.

Reply:
I agree that we should used URL or local storage. I can add a check for that. 

I am not sure why configuring the archive differently for the tests is not good.

Reply:
The intention I had was to have the archive set up to use the HTTP interface for the tests (that's how it's set up in the docker file).  The reason for this is because it's the more general interface (i.e. the one that would work anywhere, whereas the "local file" interface will only work if you're on the machine where the archive has local files).  (Both interfaces are tested separately in the submodule's own tests.)

I'm not sure what the logic is in the archive code, but if it sees a non-null local_read_dir and local_write_dir, it may just use that and not use the url.  In any event, by setting both a url and local dirs, we're ambiguously configuring the archive, and we don't want to do that.

What we really need for the tests is a way to find the archive's directories to make sure that the http transfers completed correctly, and to clean it up when we're done.  This is outside of the normal usage of the archive, and so should be configured outside of the configuration of the normal usage of the archive.

Reply:
Looking at the code, it's actually documented that if both archive_url and local_write_dir are set, it with write the file twice, once to local_write_dir, and once via the HTTP interface.  If local_read_dir is not None, then it will read from there rather than using the http interface.

So, you are changing the behavior of the archive generally by configuring both.  Unless we want to change how we interface with the archive in the tests, we need to leave local_read_dir and local_write_dir as None